action/envelope.go

```diff
@@ -28,12 +28,8 @@ type (
 	}
 
 	envelope struct {
-		version  uint32
-		chainID  uint32
-		nonce    uint64
-		gasLimit uint64
-		gasPrice *big.Int
-		payload  actionPayload
+		AbstractAction
+		payload actionPayload
```

Comment:
what is difference between ab in envelop and ab in payload?

Reply:
envelopeBuilder has its own copy of Ab, and in `build()`, use that to overwrites the payload's Ab
IMO this is actually one more thing we need to improve, we should remove `nonce, gas, gasPrice` etc from `NewTransfer()` b/c those values are useless, will be overwritten by `builder.SetNonce().SetGas()`

Reply:
It looks like they are duplicated. I think removing those params in envelope should be fine. 
```
envelop struct {
payload actionPayload
}
```

Reply:
yes, it can be in future PR to improve